src/Simplex/Messaging/Agent/Store/SQLite.hs

```diff
@@ -55,7 +57,12 @@ createSQLiteStore dbFilename = do
 connectSQLiteStore :: MonadUnliftIO m => String -> m SQLiteStore
 connectSQLiteStore dbFilename = do
   dbConn <- liftIO $ DB.open dbFilename
-  return SQLiteStore {dbFilename, dbConn}
+  compileOptions <- liftIO (DB.query_ dbConn "pragma COMPILE_OPTIONS;" :: IO [[T.Text]])
+  let threadsafeOption = find (isPrefixOf "THREADSAFE=") (concat compileOptions)
```

Comment:
what does it actually return - a single item result that is array of texts? Just was wondering if it should be just maybeHead or something...

Reply:
array of arrays actually, that's why I do `concat`